### PR TITLE
Implementation first version

### PR DESCRIPTION
**Pull request by targs08**
Implement composer plugin which generate `config.platform` option for running platform.

### CODE REVIEW

**Comment by tyranron** [comment]
@targs08 when creating new project you should ensure:

1. You've inited repo with `.editorconfig` and `.gitignore` files.
2. You've set up [project settings](https://github.com/instrumentisto/composer-platform-generator/settings) (disable unnecessary Wikis and stuff, configure PR settings, and so on..).

**Comment by targs08** [comment]
FCM 

```
Implementation composer plugin (#1)
```

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "instrumentisto/composer-platform-generator",
+    "type": "composer-plugin",
```
Comment (tyranron) [inline composer.json]: @targs08 version is missed.

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Generate config.platform section for composer.json",
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "Instrumentisto Team",
+            "email": "developer@instrumentisto.com"
+        }
+    ],
+    "require": {
+        "composer-plugin-api": "^1.1"
```
Comment (tyranron) [inline composer.json]: @targs08 used PHP extensions are missed (`json`, `prce`).

File: src/Command/UpdatePlatformReqsCommand.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+use Composer\Json\JsonManipulator;
+use Composer\Repository\PlatformRepository;
+
+
+/**
+ * Generate platform requirements
+ */
+class UpdatePlatformReqsCommand extends BaseCommand
+{
+    protected function configure()
```
Comment (tyranron) [inline src/Command/UpdatePlatformReqsCommand.php:24]: @targs08 PHP docs are missed.

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "instrumentisto/composer-platform-generator",
+    "type": "composer-plugin",
+    "description": "Generate config.platform section for composer.json",
+    "license": "MIT",
```
Comment (tyranron) [inline composer.json]: @targs08 `LICENSE.md` file in repo is missed.

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            "email": "developer@instrumentisto.com"
+        }
+    ],
+    "require": {
+        "composer-plugin-api": "^1.1"
+    },
+    "require-dev": {
+        "composer/composer": "*"
+    },
+    "autoload": {
```
Comment (tyranron) [inline composer.json]: @targs08 indentation doesn't consider `.editorconfig` rules.

**Review by tyranron — APPROVED** [review]
I've fixed the all stuff mention below.

Next time you should consider project creation more seriously. And make it much perfect for everyone else's uses.

At the moment this PR may be merged, but the project is not ready yet. You need one PR more for adding tests to the project and integrating them with Travis CI. After that is done, we may release `0.1.0` version and publish it to Packagist.

**Comment by tyranron** [comment]
@targs08 FCM

```
Implement command as Composer plugin (#1)

- create 'update-platform-reqs' Composer command
- register this command as Composer plugin
- describe plugin's purpose and usage in README

Additionally:
- update Git ignoring rules with Composer dirs
- add license
```

**Pull request merged**